### PR TITLE
chore: Simplify scheme to use clientgoscheme for KWOK

### PR DESCRIPTION
**Pull request by jigisha620**
Fixes #N/A 

**Description**
Consumers will automatically register Karpenter's scheme to clientgo when importing Karpenter types. This enables callers to simplify not think about Karpenter's schemes.

**How was this change tested?**
`make presubmit`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/pkg/environment/common/environment.go [hunk 1]
```diff
@@ -158,9 +158,7 @@ func (env *Environment) DefaultNodeClass() *v1alpha1.KWOKNodeClass {
 func (env *Environment) DefaultNodePool(nodeClass *v1alpha1.KWOKNodeClass) *v1beta1.NodePool {
 	nodePool := test.NodePool()
 	nodePool.Spec.Template.Spec.NodeClassRef = &v1beta1.NodeClassReference{
-		Name:       "default",
-		Kind:       "KWOKNodeClass",
-		APIVersion: v1alpha1.SchemeGroupVersion.Version,
+		Name: nodeClass.Name,
```
Comment (njtran) [inline test/pkg/environment/common/environment.go:161]: Don't we want kind and group/apiversion included?

Comment (jigisha620) [reply test/pkg/environment/common/environment.go]: Done

**Review by jonathan-innis — APPROVED** [review]
/lgtm
/approve

File: test/pkg/environment/common/environment.go [hunk 2]
```diff
@@ -157,10 +158,11 @@ func (env *Environment) DefaultNodeClass() *v1alpha1.KWOKNodeClass {
 
 func (env *Environment) DefaultNodePool(nodeClass *v1alpha1.KWOKNodeClass) *v1beta1.NodePool {
 	nodePool := test.NodePool()
+	gvk := object.GVK(nodeClass)
 	nodePool.Spec.Template.Spec.NodeClassRef = &v1beta1.NodeClassReference{
 		Name:       "default",
-		Kind:       "KWOKNodeClass",
-		APIVersion: v1alpha1.SchemeGroupVersion.Version,
+		Kind:       gvk.Kind,
+		APIVersion: gvk.Version,
```
Comment (jonathan-innis) [inline test/pkg/environment/common/environment.go:165]: This is just the version and doesn't contain the group. It should have both

Comment (jigisha620) [reply test/pkg/environment/common/environment.go]: Right, updated.

File: test/pkg/environment/common/environment.go [hunk 3]
```diff
@@ -157,10 +159,11 @@ func (env *Environment) DefaultNodeClass() *v1alpha1.KWOKNodeClass {
 
 func (env *Environment) DefaultNodePool(nodeClass *v1alpha1.KWOKNodeClass) *v1beta1.NodePool {
 	nodePool := test.NodePool()
+	gvk := object.GVK(nodeClass)
 	nodePool.Spec.Template.Spec.NodeClassRef = &v1beta1.NodeClassReference{
 		Name:       "default",
-		Kind:       "KWOKNodeClass",
-		APIVersion: v1alpha1.SchemeGroupVersion.Version,
+		Kind:       gvk.Kind,
```
Comment (jonathan-innis) [inline test/pkg/environment/common/environment.go:165]:
```suggestion
		Kind:       object.GVK(nodeClass).Kind,
```

consider inlining this -- I don't think it's expensive to do so

**Review by jonathan-innis — APPROVED** [review]
/lgtm
/approve

**Pull request merged**